massa-node/src/main.rs

```diff
@@ -166,16 +166,26 @@ async fn launch(
     })
     .expect("could not start selector worker");
 
-    // Create final state
-    let final_state = Arc::new(parking_lot::RwLock::new(
-        FinalState::new(
-            final_state_config,
-            Box::new(ledger),
-            selector_controller.clone(),
-            args.restart_from_snapshot_at_period.unwrap_or_default(),
-        )
-        .expect("could not init final state"),
-    ));
+    // Create final state, from snapshot or from scratch
+    let final_state = match args.restart_from_snapshot_at_period {
+        Some(last_start_period) => Arc::new(parking_lot::RwLock::new(
+            FinalState::from_snapshot(
```

Comment:
Might be good to rename so that it more clearly conveys that something is being created (in this case, using last_start_period as a seed-value in a way), as opposed to some sort of `from_$TYPE` conversion method.

```suggestion
            FinalState::derive_from_period(
```

Reply:
I don't think the last_start_period variable is that important here. Yes the value is useful to interpolate the downtime, but the primary goal is to recover all the values from the snapshot!

Maybe new_derived_from_snapshot()?

Reply:
That makes sense, go for it. I'll leave it to you to resolve this thread to mark that change as done.